shpc/main/modules/module.py

```diff
@@ -9,17 +9,32 @@
 
 
 class Module:
-    def __init__(self, name):
+    def __init__(self, config):
         """
         New module metadata and shared functions.
 
         This should be created by base.py new_module to ensure the same
         container and settings are carried forward here.
         """
-        self.name = name
+
+        # config can be either a string (a module name)
+        if isinstance(config, str):
+            self.name = config
+            self._uri = None
+        # or a ContainerConfig
+        else:
+            # We currently support gh, docker, path, or oras
+            uri = config.get_uri()
+            # If we have a path, the URI comes from the name
+            if ".sif" in uri:
+                uri = str(config.name).split(":", 1)[0]
+            self.name = uri + ":" + config.tag.name
+            self._uri = uri
+            self.config = config
+
+        self.module_basepath = self.name.replace(":", os.sep)
 
         # Cache variable properties
-        self._uri = None
         self._container_dir = None
```

Comment:
These can be moved in a block to the top - the expected is that set_config could over-ride them.